### PR TITLE
refactor: rename mod and rem to `modulo` and `remainder`

### PR DESCRIPTION
**Pull request by jaschdoc**
Related to https://github.com/flix/flix/issues/7163

### CODE REVIEW

**Comment by magnus-madsen** [comment]
@mlutze Speedy review, just for safety?

**Comment by magnus-madsen** [comment]
@stephentetley This is a breaking change. Unfortunately we have to do it to support the new parser.

File: main/test/ca/uwaterloo/flix/library/TestInt16.flix [hunk 1]
```diff
@@ -107,43 +107,43 @@ mod TestInt16 {
     // mod                                                                     //
     /////////////////////////////////////////////////////////////////////////////
     @test
-    def mod01(): Bool = (12000i16 `Int16.mod` 2000i16) == 0i16
+    def mod01(): Bool = (12000i16 `Int16.modulo` 2000i16) == 0i16
```
Comment (mlutze) [inline main/test/ca/uwaterloo/flix/library/TestInt16.flix:110]: Change test names too?

**Comment by magnus-madsen** [comment]
@mlutze Will you update your CI build once merged? Thanks

**Comment by mlutze** [comment]
> @mlutze Will you update your CI build once merged? Thanks

Surely whoever broke it should fix it :D

But yes

**Review by mlutze — APPROVED** [review]

File: main/test/ca/uwaterloo/flix/library/TestBigInt.flix [hunk 2]
```diff
@@ -86,43 +86,43 @@ mod TestBigInt {
     // mod                                                                     //
     /////////////////////////////////////////////////////////////////////////////
     @test
-    def mod01(): Bool = (1200000000000000000000ii `BigInt.mod` 200000000000000000000ii) == 0ii
+    def mod01(): Bool = (1200000000000000000000ii `BigInt.modulo` 200000000000000000000ii) == 0ii
```
Comment (magnus-madsen) [inline main/test/ca/uwaterloo/flix/library/TestBigInt.flix:89]: @jaschdoc Test case names. Easy fix in subsequent PR.

**Comment by stephentetley** [comment]
Thanks for the heads up Magnus.

**Pull request merged**